cmake/CMakeLists.txt

```diff
@@ -432,7 +432,11 @@ if(NOT WIN32)
 	LINK_DIRECTORIES(${LINK_DIRECTORIES} /usr/lib /usr/lib64 /usr/local/lib /usr/local/lib64)
 endif()
 
-SET(IFCOPENSHELL_LIBRARIES IfcParse IfcGeom)
+if(BUILD_SHARED_LIBS)
```

Comment:
You see except IfcParseExamples all targets will use either both libraries (IfcParse and IfcGeom) or none of them. To have only one library to load all should be in IfcParse.

$ grep -r -n -e "TARGET_LINK_LIBRARIES" -A1
cmake/CMakeLists.txt:459:       TARGET_LINK_LIBRARIES(IfcParse ${ICU_LIBRARIES} ${Boost_LIBRARIES})
cmake/CMakeLists.txt-460-ENDIF()
--
cmake/CMakeLists.txt:473:TARGET_LINK_LIBRARIES(IfcGeom IfcParse)
cmake/CMakeLists.txt-474-
--
cmake/CMakeLists.txt:495:TARGET_LINK_LIBRARIES(IfcConvert ${IFCOPENSHELL_LIBRARIES} ${OPENCASCADE_LIBRARIES} ${Boost_LIBRARIES} ${OPENCOLLADA_LIBRARIES} ${ICU_LIBRARIES})
cmake/CMakeLists.txt-496-if (NOT WIN32)
--
cmake/CMakeLists.txt:505:TARGET_LINK_LIBRARIES(IfcGeomServer ${IFCOPENSHELL_LIBRARIES} ${OPENCASCADE_LIBRARIES} ${Boost_LIBRARIES} ${ICU_LIBRARIES})
cmake/CMakeLists.txt-506-if (NOT WIN32)
--
src/examples/CMakeLists.txt:21:TARGET_LINK_LIBRARIES(IfcParseExamples IfcParse)
src/examples/CMakeLists.txt-22-set_target_properties(IfcParseExamples PROPERTIES FOLDER Examples)
--
src/examples/CMakeLists.txt:25:TARGET_LINK_LIBRARIES(IfcOpenHouse ${IFCOPENSHELL_LIBRARIES} ${OPENCASCADE_LIBRARIES})
src/examples/CMakeLists.txt-26-set_target_properties(IfcOpenHouse PROPERTIES FOLDER Examples)
--
src/ifcmax/CMakeLists.txt:31:TARGET_LINK_LIBRARIES(IfcMax ${IFCLIBS} Comctl32.lib zlibdll.lib bmm.lib core.lib CustDlg.lib edmodel.lib expr.lib
src/ifcmax/CMakeLists.txt-32-    flt.lib geom.lib gfx.lib gup.lib imageViewers.lib ManipSys.lib maxnet.lib Maxscrpt.lib
--
src/qtviewer/CMakeLists.txt:55:TARGET_LINK_LIBRARIES (QTviewer ${IFCLIBS} ${QT_LIBRARIES} ${OPENCASCADE_LIBRARIES})